zng/split.go

```diff
@@ -0,0 +1,22 @@
+package zng
+
+import (
+	"fmt"
+
+	"github.com/brimsec/zq/zcode"
+)
+
+func Split(elemType Type, b zcode.Bytes) ([]Value, error) {
+	if b == nil {
+		return nil, nil
+	}
+	vals := []Value{}
+	for it := b.Iter(); !it.Done(); {
+		zv, _, err := it.Next()
+		if err != nil {
+			return nil, fmt.Errorf("parsing element type '%s' value %q: %w", elemType.ZSON(), zv, err)
+		}
+		vals = append(vals, Value{elemType, zv})
+	}
+	return vals, nil
+}
```

Comment:
If you do this, then a zng empty array is returned instead of a nil array and if you change vals as you suggest, then nil array gets returned for empty array.   I got bit by this in the past...

Reply:
I'll add a comment.